app/scripts/modules/core/src/managed/ManagedReader.ts

```diff
@@ -82,16 +100,41 @@ export class ManagedReader {
     return response;
   }
 
+  private static sortEnvironments(response: IManagedApplicationEnvironmentSummary) {
+    // TODO: move this function to keel or optimize it and write some tests
+    try {
+      response.artifacts.map((artifact) => {
+        artifact.versions.map((version) => {
+          version.environments.sort((a, b) => {
+            if (isEnvDependsOn(a, b)) {
+              return -1;
+            } else if (isEnvDependsOn(b, a)) {
+              return 1;
+            }
+            // Prioritize if has any depends on
+            if (envHasAnyDependsOn(a)) return -1;
+            if (envHasAnyDependsOn(b)) return 1;
```

Comment:
Aren't these backwards? If `a` has any dependencies, it needs to come later in the list, and so is "greater".

Reply:
Nope, the ones with dependencies should come first.